app/controllers/posts_controller.rb

```diff
@@ -2,9 +2,81 @@ class PostsController < ApplicationController
   def index
     @posts = Post.where(author_id: params[:user_id])
     @user_data = User.find(params[:user_id])
+    @current_user_id = current_user.id
+  end
+
+  def new_comment
+    comment = Comment.new
+    respond_to do |format|
+      format.html { render :new_comment, locals: { comment: } }
+    end
+  end
+
+  def create_comment
+    comment_text = params.require(:comment).permit(:text)[:text]
+    post = Post.where(author_id: current_user.id).where(id: params[:id])[0]
+    user = User.find(current_user.id)
+    new_comment = Comment.new(post:, author: user, text: comment_text)
+
+    new_comment.save
+    puts new_comment.errors.full_messages
+    if new_comment.save
+      new_comment.update_comments_counter
+      flash[:success] = 'Success: Comment created successfully'
+      redirect_to user_posts_path(user_id: current_user.id, id: post.id)
+    else
+      flash.now[:error] = 'Error: Comment creation failed'
+      render :show, locals: { post: }
+    end
+  end
```

Comment:
Great job @yohannesdagnachew 

- [ ] Please create a `controllers/comment_controller.rb` file to handle comment actions.
- [ ] Please there should be a pair of routes in the `config/routes.rb` file for creating comments.
- [ ] There should be a form in the `views/comments/new.html.erb` file which accepts input for creating comments.
- [ ] Please make use of the [strong parameters](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/Controllers_lesson.md) in your `create_comment` method as explain in the `post_create` method below.